Documentation/contributing.rst

```diff
@@ -19,20 +19,115 @@ You need to have the following tools available in order to effectively
 contribute to Cilium:
 
 - git
+- go
+  [https://golang.org/dl/]
 - go-swagger
-  `go get -u github.com/go-swagger/go-swagger/cmd/swagger`
+  ``go get -u github.com/go-swagger/go-swagger/cmd/swagger``
 - go-bindata
-  `go get -u github.com/jteeuwen/go-bindata/...`
+  ``go get -u github.com/jteeuwen/go-bindata/...``
+- VirtualBox (if not using libvirt)
+  [https://www.virtualbox.org/wiki/Downloads]
+- Vagrant
+  [https://www.vagrantup.com/downloads.html]
 
-Testsuite
----------
+You should start with the Getting Started Guide
+[http://cilium.readthedocs.io/en/latest/gettingstarted/], which walks
+you through the set-up, such as installing Vagrant, getting the Cilium
+sources, and going through some Cilium basics.
+
+  
+Vagrant Setup
+~~~~~~~~~~~~~
+
+While the the Getting Started Guide
+[http://cilium.readthedocs.io/en/latest/gettingstarted/] uses a
+Vagrantfile tuned for the basic walk through, the setup for the
+Vagrantfile in the root of the Cilium tree depends on a number of
+environment variables and network setup that are managed via the
+``contrib/vagrant/start.sh``.
+
+Your Cilium tree is mapped to the VM so that you do not need to keep
+copying files between your host and the VM.  The default sync method
+is rsync, which only syncs when the VM is brought up, or when manually
+triggered (``rsync`` command in the Cilium tree).  You can also use
+NFS to access your Cilium tree from the VM by setting the environment
+variable `NFS` before running the startup script (``export NFS=1``).
+Note that your host firewall have the NFS UDP ports open, the startup
+script will give the address and port details for this.
+
+.. note::
+
+   OSX file system is by default case insensitive, which can confuse
+   git.  At the writing of this Cilium repo has no file names that
+   would be considered referring to the same file on a case
+   insensitive file system.  Regardless, it may be useful to create a
+   disk image with a case sensitive file system for holding your git
+   repos.
+
+.. note::
+
+   VirtualBox for OSX currently (version 5.1.22) always reports
+   host-only networks' prefix length as 64.  Cilium needs this prefix
+   to be 16, and the startup script will check for this.  This check
+   always fails when using VirtualBox on OSX, but it is safe to let
+   the startup script to reset the prefix length to 16.
+
+With the caveats above, starting up the build/test VM is done with:
+
+``$ ./contrib/vagrant/start.sh``
+
```

Comment:
I would drop the \`\` ones and use :: instead to get the frame rendering. For a example see https://github.com/cilium/cilium/blame/master/Documentation/gettingstarted.rst#L49-L51